frontend/src/components/PredictionTools.jsx

```diff
@@ -1,6 +1,50 @@
 import LineChart from "./predictiontools/LineChart";
+import { useState } from "react";
+import { BASE_URL } from "../lib/utils";
+import { useEffect } from "react";
+
+const PredictionTools = ({
+  portfolioData,
+  companiesData,
+  companiesStockData,
+}) => {
+  const [realData, setRealData] = useState(null);
+  const [predictionsClicked, setPredictionsClicked] = useState(false);
+  const [sum, setSum] = useState(0);
+  async function getMode() {
+    setPredictionsClicked(true);
+    const response = await fetch(
+      `${BASE_URL}/portfolios/model/${portfolioData.id}`,
+      {
+        method: "POST",
+        credentials: "include",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          currentPrice: sum,
+        }),
+      }
+    );
+    const data = await response.json();
+    setRealData(data);
+    setPredictionsClicked(false);
+  }
+  const sumValues = () => {
+    if (companiesStockData == null) {
+      return;
+    }
+    let sum = 0;
+    for (let val of companiesStockData) {
+      sum += val.price;
+    }
+    setSum(sum);
+  };
+
+  useEffect(() => {
+    sumValues();
+  }, [companiesStockData]);
```

Comment:
this is just a static calculation? you can avoid `useEffect` and creating a new state by using [`useMemo`](https://react.dev/reference/react/useMemo) instead

```
const sum = useMemo(() => { .... }, [companiesStockData]);
```